### PR TITLE
allow setting stage to false for transform steps that dont use input or output nodes

### PR DESCRIPTION
**Pull request by cliu587**
PTAL @sb2nov, @darinyu-coursera. Will land after the new load_reload_pk step is tested and with this diph.

### CODE REVIEW

**Comment by cliu587** [comment]
@sb2nov do you know why we set `output_node=base_output_node` at https://github.com/coursera/dataduct/pull/179/files#diff-59074e91ee415f9f629abf53692c99b4L114?

It seems `self._output` is potentially different from `base_output_node`as per the computation in L103.

File: dataduct/steps/transform.py [hunk 1]
```diff
@@ -53,20 +57,18 @@ def __init__(self,
...
                 'Only one of script, command and directory allowed')
 
-        base_output_node = None
-        if not no_output:
-            # Create output_node based on output_path
-            base_output_node = self.create_s3_data_node(
-                self.get_output_s3_path(get_modified_s3_path(output_path)))
+        # Create output_node based on output_path
+        base_output_node = self.create_s3_data_node(
+            self.get_output_s3_path(get_modified_s3_path(output_path)))
```
Comment (sb2nov) [inline dataduct/steps/transform.py]: why remove the if as it will create a useless S3Node, doesn't make an actual difference

**Comment by sb2nov** [comment]
If we used self_output it will create multiple staging directories instead what we want is a single staging directory that gets mapped to multiple nodes based on subdirectories so that the command doesn't need to figure out which staging directory maps to what output and is easier to manage.

**Comment by sb2nov** [comment]
Let me know if you want more details on it.

**Comment by sb2nov** [comment]
LGTM though

**Pull request merged**